tests/plugins/test_accounts.py

```diff
@@ -158,6 +180,170 @@ def test_authentication_refresh_the_cache_each_time_we_authenticate(self):
+            self.app.post_json(
+                "/accounts",
+                {
+                    "data": {
+                        "id": "alice@example.com",
+                        "password": "12éé6",
+                        "activation-form-url": "https://example.com",
+                    }
+                },
+                status=201,
+            )
+        resp = self.app.post_json(
+            "/accounts/alice@example.com/validate/" + uuid_string, {}, status=200
+        )
+        assert "activation-form-url" not in resp.json
+        assert "activation-key" not in resp.json
+        # An active user can authenticate.
+        resp = self.app.get("/", headers=get_user_headers("alice@example.com", "12éé6"))
+        assert resp.json["user"]["id"] == "account:alice@example.com"
+
+    def test_validation_fail_active_user(self):
+        uuid_string = "20e81ab7-51c0-444f-b204-f1c4cfe1aa7a"
+        with mock.patch("uuid.uuid4", return_value=uuid.UUID(uuid_string)):
+            self.app.post_json(
+                "/accounts",
+                {
+                    "data": {
+                        "id": "alice@example.com",
+                        "password": "12éé6",
+                        "activation-form-url": "https://example.com",
+                    }
+                },
+                status=201,
+            )
+        # Validate the user.
+        resp = self.app.post_json(
+            "/accounts/alice@example.com/validate/" + uuid_string, {}, status=200
+        )
+        assert "activation-form-url" not in resp.json
+        assert "activation-key" not in resp.json
+        # Try validating again.
+        resp = self.app.post_json(
+            "/accounts/alice@example.com/validate/" + uuid_string, {}, status=403
+        )
+        assert "Account alice@example.com has already been validated" in resp.json["message"]
+
+    def test_dont_check_activation_form_url_on_an_active_user(self):
+        # Once the user is active, don't check for the activation-form-url or add an activation-key.
+        uuid_string = "20e81ab7-51c0-444f-b204-f1c4cfe1aa7a"
+        with mock.patch("uuid.uuid4", return_value=uuid.UUID(uuid_string)):
+            self.app.post_json(
+                "/accounts",
+                {
+                    "data": {
+                        "id": "alice@example.com",
+                        "password": "12éé6",
+                        "activation-form-url": "https://example.com",
+                    }
+                },
+                status=201,
+            )
+        self.app.post_json("/accounts/alice@example.com/validate/" + uuid_string, {}, status=200)
+        resp = self.app.patch_json(
+            "/accounts/alice@example.com",
+            {"data": {"some-other-metadata": "foobar"}},
+            status=200,
+            headers=get_user_headers("alice@example.com", "12éé6"),
+        )
+        print(resp)
```

Comment:
Oh!

Reply:
errr... those are not the debug messages you're looking for

Reply:
Ah.